src/webui/websearchhandler.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2018  Thomas Piccirello <thomas.piccirello@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "websearchhandler.h"
+
+#include "base/bittorrent/session.h"
+#include "base/search/searchdownloadhandler.h"
+
+WebSearchHandler::WebSearchHandler(SearchHandler *searchHandler)
+    : m_searchHandler(searchHandler)
+{
+    connect(searchHandler, &SearchHandler::newSearchResults, this, &WebSearchHandler::appendSearchResults);
+    // connect(searchHandler, &SearchHandler::searchFinished, this, &WebSearchHandler::searchFinished);
+    // connect(searchHandler, &SearchHandler::searchFailed, this, &WebSearchHandler::searchFailed);
+    connect(this, &QObject::destroyed, searchHandler, &QObject::deleteLater);
+}
+
+bool WebSearchHandler::isActive()
+{
+    return m_searchHandler->isActive();
+}
+
+void WebSearchHandler::cancelSearch()
+{
+    clearQueue();
+    m_searchHandler->cancelSearch();
+}
+
+QList<SearchResult> WebSearchHandler::getSearchResults()
+{
+    QMutexLocker lock(&m_stdoutQueueMutex);
+    const int size = m_stdoutQueue.size();
+    QList<SearchResult> searchResultList;
+
+    // limit to 500 search results
+    for (int i = 0; (i < size) && (i < 500); i++)
+        searchResultList << m_stdoutQueue.dequeue();
```

Comment:
QQueue inherits from QList, if you used `QList m_stdoutQueue` instead, you can just `return m_stdoutQueue;` below.

Reply:
Or if you can predict some upper bound of the size of `m_stdoutQueue`, you can use `QVector` + `QVector::reserve()`

Reply:
> QQueue inherits from QList, if you used QList m_stdoutQueue instead, you can just return m_stdoutQueue; below.

I limit the number of results to 500 (arbitrary number) so that the response isn't too large. Though I'm switching to using `QList::mid()` for this.